tty.c

```diff
@@ -2385,6 +2404,54 @@ tty_check_bg(struct tty *tty, struct window_pane *wp, struct grid_cell *gc)
 		gc->bg -= 90;
 }
 
+static void
+tty_check_us(struct tty *tty, struct window_pane *wp, struct grid_cell *gc)
+{
+	u_char	r, g, b;
+	u_int	colours;
+	int	c;
+
+	/* Perform substitution if this pane has a palette. */
+	if (~gc->flags & GRID_FLAG_NOPALETTE) {
+		if ((c = window_pane_get_palette(wp, gc->us)) != -1)
+			gc->us = c;
+	}
+
+	/* Check if this is a 24-bit colour. */
+	if (gc->us & COLOUR_FLAG_RGB) {
+		/* Not a 24-bit terminal? Translate to 256-colour palette. */
+		if (!tty_term_has(tty->term, TTYC_SETRGBU)) { // TODO: kmo
```

Comment:
If I understood this correctly, `TTYC_SETRGBU` (which I simply made up) would be another termcap, that specifies whether the target terminal can set the underline color in 24-bit mode. What termcap should I use here?

Reply:
I would only have Smulc which sets RGB colour and map the 256 colour entry to the nearest RGB colour. I don't think we need two terminfo capabilities for this.